tests/plugins/test_update_parent_image_stream_tag.py

```diff
@@ -0,0 +1,125 @@
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import unicode_literals
+
+from atomic_reactor.core import DockerTasker
+from atomic_reactor.inner import DockerBuildWorkflow
+from atomic_reactor.plugin import PreBuildPluginsRunner, PluginFailedException
+from atomic_reactor.plugins.pre_update_parent_image_stream_tag import \
+                                             UpdateParentImageStreamTagPlugin
+from atomic_reactor.util import ImageName
+from flexmock import flexmock
+from osbs.api import OSBS
+from osbs.exceptions import OsbsResponseException
+from tests.constants import SOURCE, MOCK
+
+import json
+import osbs.conf
+import pytest
+
+if MOCK:
+    from tests.docker_mock import mock_docker
+
+
+class X(object):
+    pass
+
+
+class TestUpdateParentImageStreamTag(object):
+    def prepare(self):
+        if MOCK:
+            mock_docker()
+        tasker = DockerTasker()
+        workflow = DockerBuildWorkflow(SOURCE, "test-image")
+        setattr(workflow, 'builder', X())
+        setattr(workflow.builder, 'image_id', 'asd123')
+        setattr(workflow.builder, 'base_image', ImageName(repo='Fedora',
+                                                          tag='22'))
+        setattr(workflow.builder, 'source', X())
+        setattr(workflow.builder.source, 'path', '/tmp')
+        setattr(workflow.builder.source, 'dockerfile_path', None)
+
+        return workflow, tasker
+
+    @pytest.mark.parametrize(('status_code'), (200, 404, 500))
+    @pytest.mark.parametrize(('changed'), (True, False))
+    @pytest.mark.parametrize(('scheduled'), (True, False, None))
+    def test_update_parent_image_stream_tag_plugin(self, monkeypatch,
+                                                   status_code, changed,
+                                                   scheduled):
+        workflow, tasker = self.prepare()
+
+        openshift_url = 'http://openshift-url.com'
+        namespace = 'namespace1'
+        image_stream_id = 'imagestream1'
+        image_stream_tag_name = 'imagestreamtag1'
+        image_stream_tag_id = 'imagestream1:imagestreamtag1'
+
+        image_stream = {'kind': 'ImageStream'}
+        image_stream_response = flexmock()
+        image_stream_response.should_receive('json').and_return(image_stream)
+
+        _osbs = flexmock(OSBS)
+
+        get_image_stream = (_osbs.should_receive('get_image_stream')
+                            .with_args(image_stream_id)
+                            .once())
+
+        if status_code == 200:
+            get_image_stream.and_return(image_stream_response)
+
+            (_osbs.should_receive('ensure_image_stream_tag')
+             .with_args(image_stream, image_stream_tag_name, bool(scheduled))
+             .once()
+             .and_return(changed))
+
+        else:
+            (get_image_stream
+             .and_raise(OsbsResponseException('error', status_code))
+             .and_return(None))
```

Comment:
I think that says 'raise an exception the first time it is called, then return None subsequently'.